### PR TITLE
Write a test for HTMLDialogElement#open

### LINKED ISSUE #12574: Write a test for HTMLDialogElement#open
Spec: https://html.spec.whatwg.org/multipage/forms.html#dom-dialog-open
Code: `components/script/dom/htmldialogelement.rs`, `components/script/dom/webidls/HTMLDialogElement.webidl`
Test: None yet, will need to write one (`./mach create-wpt tests/wpt/web-platform-tests/html/semantics/interactive-elements/the-dialog-element/dialog-open.html`)

### PR DESCRIPTION
**Pull request by vrod**
## Write a test for HTMLDialogElement#open.

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #12574.
- [X] There are tests for these changes

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/12598)

### CODE REVIEW

File: tests/wpt/web-platform-tests/html/semantics/interactive-elements/the-dialog-element/dialog-open.html [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+  test(function(){
+    assert_false(d1.open);
+    assert_true(d2.open);
+  }, "On getting, the IDL open attribute must return true if the content open attribute is set, and false if it is absent.");
+
+  test(function(){
+    d1.open = true;
+    assert_true(d1.hasAttribute("open"));
+    assert_equals(d1.getAttribute("open"), "");
```
Comment (KiChjang) [inline tests/wpt/web-platform-tests/html/semantics/interactive-elements/the-dialog-element/dialog-open.html]: This one doesn't make sense here, I'd rather you remove this line.

**Comment by vrod** [comment]
@KiChjang: removed the line that doesn't make sense and updated commit.

**Comment by vrod** [comment]
@KiChjang: Also, I had forgotten to add a title and reference. It's there now.

File: tests/wpt/web-platform-tests/html/semantics/interactive-elements/the-dialog-element/dialog-open.html [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    assert_false(d1.open);
+    assert_true(d2.open);
+  }, "On getting, the IDL open attribute must return true if the content open attribute is set, and false if it is absent.");
+
+  test(function(){
+    d1.open = true;
+    assert_true(d1.hasAttribute("open"));
+    d2.open = false;
+    assert_false(d2.hasAttribute("open"));
+  }, "On setting, the content open attribute must be removed if the IDL open attribute is set to false, and must be set to the empty string if the IDL open attribute is set to true.");
```
Comment (KiChjang) [inline tests/wpt/web-platform-tests/html/semantics/interactive-elements/the-dialog-element/dialog-open.html]: The corresponding message here also needs to be updated.

**Comment by KiChjang** [comment]
@bors-servo r+

Thanks!

**Pull request merged**